server/src/termdb.violin.js

```diff
@@ -51,8 +51,11 @@ q = {
 
 export async function trigger_getViolinPlotData(q, res, ds, genome) {
 	const term = q.term?.term ? q.term.term : ds.cohort.termdb.q.termjsonByOneid(q.termid)
-	if (!q.term.q?.mode) q.term.q.mode = 'continuous'
 	if (!term) throw '.termid invalid'
+
+	if (!q.term.q) throw 'q.term.q{} missing'
+	q.term.q.mode = 'continuous' // override mode in case requests by discrete. which breaks gdc term
```

Comment:
should be done on a copy of the `term.q`?

Reply:
yes should